prow/test/integration/test/webhook_server_test.go

```diff
@@ -55,11 +66,73 @@ func TestWebhookServerValidateCluster(t *testing.T) {
 	//prowjob should be created successfully
 	err = exec.Command("kubectl", "--context=kind-kind-prow-integration", "apply", "-f", absValidProwJobPath).Run()
 	if err != nil {
-		t.Errorf("prowjob was not properly validated")
+		t.Errorf("prowjob was not properly validated %v", err)
 	}
 	//cleanup prowjob that was created
 	err = cleanUpCluster(validProwJobID)
 	if err != nil {
 		t.Errorf("could not delete prowjob %s", validProwJobID)
 	}
 }
+
+func TestWebhookServerMutateProwjob(t *testing.T) {
+	validBareProwJobPath := "./testdata/valid_bare_prowjob.yaml"
+	absValidBareProwJobPath, err := filepath.Abs(validBareProwJobPath)
+	var mutatedProwJob v1.ProwJob
+	if err != nil {
+		t.Errorf("could not find absolute file path")
+	}
+	//prowjob should be created successfully
+	err = exec.Command("kubectl", "--context=kind-kind-prow-integration", "apply", "-f", absValidBareProwJobPath).Run()
+	if err != nil {
+		t.Errorf("prowjob was not properly validated %v", err)
+	}
+
+	key := types.NamespacedName{
+		Namespace: "default",
+		Name:      validBareProwJobID,
+	}
+	cfg := config.GetConfigOrDie()
+	if err != nil {
+		t.Errorf("Error getting kubeconfig")
+	}
+	client, err := ctrlruntimeclient.New(cfg, ctrlruntimeclient.Options{})
+	if err != nil {
+		t.Errorf("Could not create writer client %v", err)
+	}
+	ctx := context.Background()
+	if err := client.Get(ctx, key, &mutatedProwJob); err != nil {
+		t.Errorf("could not get prowjob: %v", err)
+	}
+
+	originalProwJobYAML, err := ioutil.ReadFile(absValidBareProwJobPath)
+	if err != nil {
+		t.Errorf("unable to read yaml file %v", err)
+	}
+	var originalProwJob v1.ProwJob
+	err = yaml.Unmarshal(originalProwJobYAML, &originalProwJob)
+	if err != nil {
+		t.Errorf("unable to unmarshal into prowjob %v", err)
+	}
+	
+	origDecConfigBytes, err := json.Marshal(originalProwJob.Spec.DecorationConfig)
+	if err != nil {
+		t.Errorf("unable to marshal original prowjob decoration config %v", err)
+	}
+
+	mutatedDecConfigBytes, err := json.Marshal(mutatedProwJob.Spec.DecorationConfig)
+	if err != nil {
+		t.Errorf("unable to marshal mutated prowjob decoration config %v", err)
+	}
+
+	if bytes.Equal(origDecConfigBytes, mutatedDecConfigBytes) {
```

Comment:
instead of comparing them as bytes, might be easier to use cmp.Diff on their structs. You can get a better error message that shows what is missing from the returned prowjob